src/sentry/integrations/vsts/client.py

```diff
@@ -199,9 +201,19 @@ def _refresh_auth_if_expired(self):
         if time_expires is None:
             raise InvalidIdentity("VstsApiClient requires identity with specified expired time")
         if int(time_expires) <= int(time()):
-            self.identity.get_provider().refresh_identity(
-                self.identity, redirect_url=self.oauth_redirect_url
-            )
+            # TODO(iamrajjoshi): Remove this after migration
+            # Need this here because there is no way to get any identifier which would tell us which method we should use to refresh the token
+            from sentry.identity.vsts.provider import VSTSNewIdentityProvider
+
+            try:
+                self.identity.get_provider().refresh_identity(
+                    self.identity, redirect_url=self.oauth_redirect_url
+                )
+            except ApiError as e:
```

Comment:
when/how can we be confident we can always use the new identity provider?

Reply:
when all users are migrated to the new integration, we can updating the existing provider and remove this try catch

Reply:
ok i guess we just track a redash query